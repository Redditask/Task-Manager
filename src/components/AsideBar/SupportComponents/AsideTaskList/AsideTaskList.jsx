import React from 'react';
import AsideTask from "../AsideTask/AsideTask";

import {CSSTransition, TransitionGroup} from "react-transition-group";

import styles from "./AsideTaskList.module.scss";
import "../../Transition.scss";

import {useSelector} from "react-redux";
import {selectSelectedTasks} from "../../../../store/selectors";

const AsideTaskList = ({setSelectedTask, setEditModalStatus}) => {
    const tasks = useSelector(selectSelectedTasks);

    return (
        <TransitionGroup>
            {tasks.length
                ? tasks.map((task,index)=>(
                        <CSSTransition
                            key = {task+index}
                            timeout={500}
                            classNames="item"
                        >
                            <AsideTask
                                task={task}
                                setTask={setSelectedTask}
                                setEditModalStatus={setEditModalStatus}
                            />
                        </CSSTransition>
                    )
                )
                : <CSSTransition
                    key="tasksIsEmpty"
                    timeout={500}
                    classNames="item"
                >
                    <h3 className={styles.TasksIsEmpty}>No task today</h3>
                </CSSTransition>
            }
        </TransitionGroup>
    );
};

export default AsideTaskList;
