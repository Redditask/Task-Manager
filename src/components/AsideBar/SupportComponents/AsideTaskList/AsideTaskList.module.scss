.TasksIsEmpty {
  font-weight: normal;
  font-size: 2.5rem;
}
