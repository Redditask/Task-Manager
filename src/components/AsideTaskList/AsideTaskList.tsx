import React from 'react';
import AsideTask from "../AsideTask/AsideTask";

import {CSSTransition, TransitionGroup} from "react-transition-group";

// @ts-ignore
import styles from "./AsideTaskList.module.scss";
import "../../Transition.scss";

import {selectSelectedTasks} from "../../store/selectors";
import {useAppSelector} from "../../hooks/hooks";

import {Task} from "../../types/data";

interface AsideTaskListProps {
    setSelectedTask: (task: Task) => void;
    setEditModalStatus: (editModalStatus: boolean) => void;
}

const AsideTaskList: React.FC<AsideTaskListProps> = ({setSelectedTask, setEditModalStatus}) => {
    const tasks: Task[] = useAppSelector(selectSelectedTasks);

    return (
        <TransitionGroup>
            {tasks.length
                ? tasks.map((task, index) => (
                        <CSSTransition
                            key={`${index}-${task.id || "undefinedId"}`}
                            timeout={500}
                            classNames="item"
                        >
                            <AsideTask
                                task={task}
                                setTask={setSelectedTask}
                                setEditModalStatus={setEditModalStatus}
                            />
                        </CSSTransition>
                    )
                )
                : <CSSTransition
                    key="tasksIsEmpty"
                    timeout={500}
                    classNames="item"
                >
                    <h3 className={styles.TasksIsEmpty}>No task today</h3>
                </CSSTransition>
            }
        </TransitionGroup>
    );
};

export default AsideTaskList;
