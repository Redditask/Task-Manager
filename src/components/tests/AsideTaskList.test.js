import {render} from "@testing-library/react";
import * as ReduxHooks from "react-redux";

import AsideTaskList from "../AsideBar/SupportComponents/AsideTaskList/AsideTaskList";

jest.mock("react-redux");

const mockedUseSelector = jest.spyOn(ReduxHooks, "useSelector");

const someTasks = [
    {
        id: "2021-11-27T16:33:22.812Z",
        taskText: "Your task",
        year: 2022,
        month: 5,
        day: 17,
        color: "beige"
    },
    {
        id: "2022-5-17T16:33:22.812Z",
        taskText: "Your task",
        year: 2022,
        month: 5,
        day: 17,
        color: "#00FF7F"
    }
];

describe("AsideTaskList", ()=>{
    it("should create TaskList with empty task list", ()=>{
        mockedUseSelector.mockReturnValue([]);

        // eslint-disable-next-line testing-library/render-result-naming-convention
        const component = render(<AsideTaskList/>);

        expect(component).toMatchSnapshot();
    });

    it("should create TaskList with tasks", ()=>{
        mockedUseSelector.mockReturnValue(someTasks);

        // eslint-disable-next-line testing-library/render-result-naming-convention
        const component = render(
            <AsideTaskList
                setEditModalStatus={jest.fn()}
                setSelectedTask={jest.fn()}
            />
        );

        expect(component).toMatchSnapshot();
    });
});
