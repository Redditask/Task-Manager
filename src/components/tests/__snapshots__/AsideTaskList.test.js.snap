// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsideTaskList should create TaskList with empty task list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h3
          class="TasksIsEmpty"
        >
          No task today
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h3
        class="TasksIsEmpty"
      >
        No task today
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AsideTaskList should create TaskList with tasks 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="Task"
        >
          <li
            class="Content"
          >
            <div
              class="Content__color"
              style="background: beige;"
            />
            <div>
              Your task
            </div>
          </li>
          <div
            class="ButtonsArea"
          >
            <div
              class="Button"
              title="Edit"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                />
              </svg>
            </div>
            <div
              class="Button"
              title="Remove"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="Task"
        >
          <li
            class="Content"
          >
            <div
              class="Content__color"
              style="background: rgb(0, 255, 127);"
            />
            <div>
              Your task
            </div>
          </li>
          <div
            class="ButtonsArea"
          >
            <div
              class="Button"
              title="Edit"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                />
              </svg>
            </div>
            <div
              class="Button"
              title="Remove"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="Task"
      >
        <li
          class="Content"
        >
          <div
            class="Content__color"
            style="background: beige;"
          />
          <div>
            Your task
          </div>
        </li>
        <div
          class="ButtonsArea"
        >
          <div
            class="Button"
            title="Edit"
          >
            <svg
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
              />
            </svg>
          </div>
          <div
            class="Button"
            title="Remove"
          >
            <svg
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="Task"
      >
        <li
          class="Content"
        >
          <div
            class="Content__color"
            style="background: rgb(0, 255, 127);"
          />
          <div>
            Your task
          </div>
        </li>
        <div
          class="ButtonsArea"
        >
          <div
            class="Button"
            title="Edit"
          >
            <svg
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
              />
            </svg>
          </div>
          <div
            class="Button"
            title="Remove"
          >
            <svg
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
